[
    [
        {
            "function_name": "view",
            "file_path": "/PycharmProjects/PyVul/src/context/dataset/test_data/CVE-2019-7548/discord-bot-journal-master/db.py",
            "line_number": 86,
            "source_code": "    def view(self, table_name):\n        table = self.table_mapping.get(table_name)\n        if table is not None:\n            with self.engine.connect() as connection:\n                query = table.select().order_by(table.c.id.desc()).limit(5)\n                result = connection.execute(query)\n                return result.fetchall()\n        else:\n            raise ValueError(f\"Table {table_name} does not exist\")\n",
            "parameters": [],
            "focused_calls": [
                "order_by"
            ],
            "extend_calls": [],
            "conditions_for_focused_calls": {},
            "parameter_conditions": "",
            "is_vulnerability_reachable": false,
            "unique_id": "/PycharmProjects/PyVul/src/context/dataset/test_data/CVE-2019-7548/discord-bot-journal-master/db.py:view:86"
        },
        {
            "function_name": "view",
            "file_path": "/PycharmProjects/PyVul/src/context/dataset/test_data/CVE-2019-7548/discord-bot-journal-master/bot.py",
            "line_number": 274,
            "source_code": "async def view(ctx, table):\n    data = db.view(table)\n    if data:\n        for entry in data:\n            entry_str = ', '.join([f'{key}: {value}' for key, value in zip(['id', 'date', 'minutes', 'description', 'category'], entry)])\n            await ctx.send(entry_str)\n    else:\n        await ctx.send('No data found.')\n",
            "parameters": [],
            "focused_calls": [
                "view"
            ],
            "extend_calls": [
                "async def view(ctx, table):\n    data = db.view(table)\n    if data:\n        for entry in data:\n            entry_str = ', '.join([f'{key}: {value}' for key, value in zip(['id', 'date', 'minutes', 'description', 'category'], entry)])\n            await ctx.send(entry_str)\n    else:\n        await ctx.send('No data found.')"
            ],
            "conditions_for_focused_calls": {},
            "parameter_conditions": "",
            "is_vulnerability_reachable": false,
            "unique_id": "/PycharmProjects/PyVul/src/context/dataset/test_data/CVE-2019-7548/discord-bot-journal-master/bot.py:view:274"
        }
    ]
]
